src/app/component/CurrencySwitcher/CurrencySwitcher.component.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+import React, { PureComponent } from 'react';
+import { setCurrency, getCurrency } from 'Util/Currency/Current';
+import PropTypes from 'prop-types';
+import Field from 'Component/Field/Field.container';
+
+class CurrencySwitcherComponent extends PureComponent {
+    static propTypes = {
+        setCurrencyData: PropTypes.func.isRequired,
+        currency: PropTypes.object.isRequired
+    };
+
+    changeCurrency(value) {
+        if (value !== getCurrency()) {
+            setCurrency(value);
+            window.location.reload();
+        }
+    }
+
+    renderCurrencyList() {
+        const { currency } = this.props;
+        const currencyCodes = currency.currency_codes ? currency.currency_codes : [];
+        const availableCurrency = currencyCodes.filter(elem => elem.rate > 0);
+        const selectOptions = [];
+        availableCurrency.map((currency) => {
```

Comment:
Please do not iterate the array twice. Please use reduce to both stripe out the `rate <= 0` items + collect the `selectOptions`. Preferably, something like this:
```javascript
const selectOptions = availableCurrency.reduce((acc, currency) => {
    const { currency_to, rate } = currency;

    if (rate <= 0) return acc;

    return [
        ...acc,
        {
            id: currency_to,
            label: currency_to,
            value: currency_to
        }
    ];
}, []);
```

I would even prefer moving this to container as business logic part.